internal/woc_client/woc_client.go

```diff
@@ -259,13 +236,12 @@ func (w *WocClient) GetBalanceWithRetries(ctx context.Context, mainnet bool, add
 	return balanceRes.confirmed, balanceRes.unconfirmed, nil
 }
 
-func (w *WocClient) TopUp(ctx context.Context, mainnet bool, address string) error {
-	net := "test"
-	if mainnet {
+func (w *WocClient) TopUp(ctx context.Context, address string) error {
+	if w.net != "test" {
 		return errors.New("top up can only be done on testnet")
 	}
 
-	req, err := http.NewRequestWithContext(ctx, "GET", fmt.Sprintf("https://api-test.whatsonchain.com/v1/bsv/%s/faucet/send/%s", net, address), nil)
+	req, err := http.NewRequestWithContext(ctx, "GET", fmt.Sprintf("https://api-test.whatsonchain.com/v1/bsv/test/faucet/send/%s", address), nil)
```

Comment:
It is the only method where api url is different (`api**-test**`. not `api.`)

Reply:
@pawellewandowski98 I think you're right, because this URL also works for the fauced: `https://api.whatsonchain.com/v1/bsv/test/faucet/send`

So the function `w.httpRequest()` could be used